pkg/data/cloud/platform_test.go

```diff
@@ -123,6 +123,10 @@ func TestByName(t *testing.T) {
 			name: "aws-hcp",
 			want: AWSHCP,
 		},
```

Comment:
Can we also add a test case for an empty string for `name`? Should return an empty `Platform{}`

Reply:
Good call, that failed so added a fix for that scenario as well.